### PR TITLE
[SPARK-41351][CONNECT] Column should support != operator

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?
This patch fixes a bug that the `Column` does not support the != operator.

### Why are the changes needed?
Compat

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 1]
```diff
@@ -870,6 +870,11 @@ def test_agg_with_avg(self):
         self.assertEqual(4.0, res[0][1])
         self.assertEqual(5.0, res[1][1])
 
+    def test_column_operator(self):
```
Comment (amaliujia) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:873]: nit: maybe we should move all such tests to `test_connect_functions.py`?

Comment (HyukjinKwon) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:873]: or create one `test_connect_columns.py`

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/tests/connect/test_connect_column.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# limitations under the License.
+#
+import unittest
+
+from pyspark.sql.tests.connect.test_connect_basic import SparkConnectSQLTestCase
+from pyspark.testing.sqlutils import have_pandas
+
+if have_pandas:
+    pass
+
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/connect/test_connect_column.py:24]:
```suggestion
```

Comment (HyukjinKwon) [reply python/pyspark/sql/tests/connect/test_connect_column.py:24]: Seems a mistake :-).

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**